From 53b913ce9c04f82d45268f3ba1d3c6512b2c3078 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C5=81ukasz=20Maria=C5=84ski?= <lmarianski@protonmail.com>
Date: Tue, 6 Jun 2023 22:24:38 +0000
Subject: [PATCH] Update dependencies.rs (#1667)

---
 alvr/xtask/src/dependencies.rs | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/alvr/xtask/src/dependencies.rs b/alvr/xtask/src/dependencies.rs
index 621e206f1b..830ab72d23 100644
--- a/alvr/xtask/src/dependencies.rs
+++ b/alvr/xtask/src/dependencies.rs
@@ -117,7 +117,6 @@ pub fn build_ffmpeg_linux(nvenc_flag: bool) {
         "--enable-gpl",
         "--enable-version3",
         "--enable-static",
-        "--enable-shared",
         "--disable-programs",
         "--disable-doc",
         "--disable-avdevice",
@@ -179,8 +178,6 @@ pub fn build_ffmpeg_linux(nvenc_flag: bool) {
                 "--enable-nonfree",
                 "--enable-cuda-nvcc",
                 "--enable-libnpp",
-                "--enable-hwaccel=h264_nvenc",
-                "--enable-hwaccel=hevc_nvenc",
                 "--nvccflags=\"-gencode arch=compute_52,code=sm_52 -O2\"",
                 &format!("--extra-cflags=\"{include_flags}\""),
                 &format!("--extra-ldflags=\"{link_flags}\""),
