From 4d7fd89f4938f6df9db3f3b9dbf4a76eb23fea5f Mon Sep 17 00:00:00 2001
From: Max Thomas <mtinc2@gmail.com>
Date: Sat, 22 Feb 2025 15:08:12 -0700
Subject: [PATCH] ffmpeg -> 7.1 (#2719)

---
 alvr/xtask/src/dependencies.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/alvr/xtask/src/dependencies.rs b/alvr/xtask/src/dependencies.rs
index b2e876e288..e8dd840e55 100644
--- a/alvr/xtask/src/dependencies.rs
+++ b/alvr/xtask/src/dependencies.rs
@@ -61,14 +61,14 @@ pub fn prepare_ffmpeg_windows(deps_path: &Path) {
     command::download_and_extract_zip(
         &format!(
             "https://github.com/BtbN/FFmpeg-Builds/releases/download/latest/{}",
-            "ffmpeg-n5.1-latest-win64-gpl-shared-5.1.zip"
+            "ffmpeg-n7.1-latest-win64-gpl-shared-7.1.zip"
         ),
         deps_path,
     )
     .unwrap();
 
     fs::rename(
-        deps_path.join("ffmpeg-n5.1-latest-win64-gpl-shared-5.1"),
+        deps_path.join("ffmpeg-n7.1-latest-win64-gpl-shared-7.1"),
         deps_path.join("ffmpeg"),
     )
     .unwrap();
