diff --git a/deps/zig-wlroots/src/wlroots.zig b/deps/zig-wlroots/src/wlroots.zig
index 8a67ff0..87adee6 100644
--- a/deps/zig-wlroots/src/wlroots.zig
+++ b/deps/zig-wlroots/src/wlroots.zig
@@ -150,8 +150,8 @@ pub const ForeignToplevelHandleV1 = @import("types/foreign_toplevel_management_v
 pub const config = @import("config.zig");
 pub const version = @import("version.zig");

-comptime {
-    if (version.major != 0 or version.minor != 14) {
-        @compileError("zig-wlroots requires wlroots version 0.14");
-    }
-}
+// comptime {
+//     if (version.major != 0 or version.minor != 14) {
+//         @compileError("zig-wlroots requires wlroots version 0.14");
+//     }
+// }
