diff --git a/ycmd/completers/go/go_completer.py b/ycmd/completers/go/go_completer.py
index e1c8d62a..23deb0a3 100644
--- a/ycmd/completers/go/go_completer.py
+++ b/ycmd/completers/go/go_completer.py
@@ -24,15 +24,7 @@ from ycmd import utils
 from ycmd.completers.language_server import language_server_completer
 
 
-PATH_TO_GOPLS = os.path.abspath( os.path.join( os.path.dirname( __file__ ),
-  '..',
-  '..',
-  '..',
-  'third_party',
-  'go',
-  'bin',
-  utils.ExecutableName( 'gopls' ) ) )
-
+PATH_TO_GOPLS = os.path.join( '/usr/bin', 'gopls' )
 
 def ShouldEnableGoCompleter( user_options ):
   server_exists = utils.FindExecutableWithFallback(
