diff --git a/ycmd/completers/rust/rust_completer.py b/ycmd/completers/rust/rust_completer.py
index 8a3c2747..b2941010 100644
--- a/ycmd/completers/rust/rust_completer.py
+++ b/ycmd/completers/rust/rust_completer.py
@@ -25,9 +25,7 @@ from ycmd.utils import LOGGER, re
 
 
 LOGFILE_FORMAT = 'ra_'
-RUST_ROOT = os.path.abspath(
-  os.path.join( os.path.dirname( __file__ ), '..', '..', '..', 'third_party',
-                'rust-analyzer' ) )
+RUST_ROOT = os.path.abspath( os.path.join( '/usr' ) )
 RA_BIN_DIR = os.path.join( RUST_ROOT, 'bin' )
 RUSTC_EXECUTABLE = utils.FindExecutable( os.path.join( RA_BIN_DIR, 'rustc' ) )
 RA_EXECUTABLE = utils.FindExecutable( os.path.join(
