diff --git a/ycmd/completers/typescript/typescript_completer.py b/ycmd/completers/typescript/typescript_completer.py
index 7ed67221..3dcf80e6 100644
--- a/ycmd/completers/typescript/typescript_completer.py
+++ b/ycmd/completers/typescript/typescript_completer.py
@@ -37,9 +37,7 @@ NO_DIAGNOSTIC_MESSAGE = 'No diagnostic for current line!'
 
 RESPONSE_TIMEOUT_SECONDS = 20
 
-TSSERVER_DIR = os.path.abspath(
-  os.path.join( os.path.dirname( __file__ ), '..', '..', '..', 'third_party',
-                'tsserver' ) )
+TSSERVER_DIR = os.path.abspath( '/usr/bin' ) 
 
 LOGFILE_FORMAT = 'tsserver_'
 
@@ -77,8 +75,7 @@ def FindTSServer( user_options_path ):
     return tsserver
   # The TSServer executable is installed at the root directory on Windows while
   # it's installed in the bin folder on other platforms.
-  for executable in [ os.path.join( TSSERVER_DIR, 'bin', 'tsserver' ),
-                      os.path.join( TSSERVER_DIR, 'tsserver' ),
+  for executable in [ os.path.join( TSSERVER_DIR, 'tsserver' ),
                       'tsserver' ]:
     tsserver = utils.FindExecutable( executable )
     if tsserver:
