diff --git a/CMakeLists.txt b/CMakeLists.txt
index 471f4a63..b9350c00 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,21 +73,7 @@ find_package(Clang REQUIRED)
 if(CLANG_LINK_CLANG_DYLIB)
   target_link_libraries(ccls PRIVATE clang-cpp)
 else()
-  target_link_libraries(ccls PRIVATE
-    clangIndex
-    clangFormat
-    clangTooling
-    clangToolingInclusions
-    clangToolingCore
-    clangFrontend
-    clangParse
-    clangSerialization
-    clangSema
-    clangAST
-    clangLex
-    clangDriver
-    clangBasic
-  )
+  target_link_libraries(ccls PRIVATE clang-cpp)
 endif()
 
 if(LLVM_LINK_LLVM_DYLIB)
