diff --git a/Makefile b/Makefile
index d413ebc0..eaa812be 100644
--- a/Makefile
+++ b/Makefile
@@ -226,7 +226,7 @@ endif
 
 INC =  -I$(SRCDIR)/3rdparty/include -I$(SRCDIR)/include
 CFLAGS   = -fPIC $(HAVE_MONGOOSE) $(INC) $(EXTRA_CFLAGS)
-CXXFLAGS = -O3 -std=c++0x -Wall -fPIC $(HAVE_MONGOOSE) $(INC) $(EXTRA_CXXFLAGS)
+CXXFLAGS = -std=c++0x -Wall -fPIC $(HAVE_MONGOOSE) $(INC) $(EXTRA_CXXFLAGS)
 LIBDIR = -L$(BUILD_ROOT)
 LIB =
 LDFLAGS = $(LDSTRIP) $(LDZDEF) $(EXTRA_LDFLAGS)
