From 0304f2c9d15cefa64182e03a6a2c8a76bd885d2e Mon Sep 17 00:00:00 2001
From: TheGreatMcPain <james@thegreatmcpain.xyz>
Date: Thu, 5 Oct 2023 22:01:14 -0500
Subject: [PATCH] Fix compile with GCC13

---
 blockalign_src/crc.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/blockalign_src/crc.cpp b/blockalign_src/crc.cpp
index 188ef65..1f9a8e5 100644
--- a/blockalign_src/crc.cpp
+++ b/blockalign_src/crc.cpp
@@ -122,6 +122,7 @@
 
 #if CRYPTOPP_BOOL_SSE4_INTRINSICS_AVAILABLE
 #include <smmintrin.h>
+#include <cstdint>
 #endif
 
 #ifndef CRYPTOPP_MS_STYLE_INLINE_ASSEMBLY
-- 
2.41.0

