diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7a167a2ad..c406137cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -405,10 +405,6 @@ if(ENABLE_RENDERDOCCMD)
     add_subdirectory(renderdoccmd)
 endif()
 
-# install documentation files
-install (FILES util/LINUX_DIST_README DESTINATION share/doc/renderdoc RENAME README)
-install (FILES LICENSE.md DESTINATION share/doc/renderdoc)
-
 message(STATUS "Enabled APIs:")
 if(ENABLE_GL)
     if(ENABLE_EGL)
diff --git a/qrenderdoc/CMakeLists.txt b/qrenderdoc/CMakeLists.txt
index 6d8fdb731..87d81cac2 100644
--- a/qrenderdoc/CMakeLists.txt
+++ b/qrenderdoc/CMakeLists.txt
@@ -284,7 +284,6 @@ install (PROGRAMS ${CMAKE_BINARY_DIR}/bin/qrenderdoc DESTINATION bin)
 install (FILES share/application-x-renderdoc-capture.svg DESTINATION share/icons/hicolor/scalable/mimetypes/)
 install (FILES share/renderdoc-icon-16x16.xpm DESTINATION share/pixmaps/)
 install (FILES share/renderdoc-icon-32x32.xpm DESTINATION share/pixmaps/)
-install (FILES share/magic DESTINATION share/doc/renderdoc)
 install (FILES share/menu DESTINATION share/menu RENAME renderdoc)
 install (FILES share/renderdoc.desktop DESTINATION share/applications)
 install (FILES share/renderdoc.thumbnailer DESTINATION share/thumbnailers)
