diff --git a/src/cmake/FindGMP.cmake b/src/cmake/FindGMP.cmake
index 96d4a55..c82d705 100644
--- a/src/cmake/FindGMP.cmake
+++ b/src/cmake/FindGMP.cmake
@@ -68,6 +68,14 @@ if(GMP_INCLUDES)
   endif()
 endif()
 
+# Fix busted gmp detection
+set (GMP_FIND_VERSION_MAJOR 6)
+set (GMP_FIND_VERSION_MINOR 2)
+set (GMP_FIND_VERSION_PATCH 1)
+  set(GMP_FIND_VERSION
+    "${GMP_FIND_VERSION_MAJOR}.${GMP_FIND_VERSION_MINOR}.${GMP_FIND_VERSION_PATCH}")
+set (GMP_VERSION_OK TRUE)
+
 find_library(GMP_LIBRARIES gmp PATHS $ENV{GMPDIR} ${LIB_INSTALL_DIR})
 
 include(FindPackageHandleStandardArgs)
