commit 5672780745ed8d4bc34badf40176043f674155bc
Author: TheGreatMcPain <james@thegreatmcpain.xyz>
Date:   Sun Jun 12 03:01:36 2022 -0500

    Allow for external config directory

diff --git a/store/constants.py b/store/constants.py
index 21b30fa..57e33df 100644
--- a/store/constants.py
+++ b/store/constants.py
@@ -1,14 +1,37 @@
 import os
 import sys
 
+
+def get_config_dir():
+    configRoot = ""
+    if "APPDATA" in os.environ:
+        configRoot = os.environ["APPDATA"]
+    elif "XDG_CONFIG_HOME" in os.environ:
+        configRoot = os.environ["XDG_CONFIG_HOME"]
+    elif os.path.isdir(os.path.expanduser("~/.config")):
+        configRoot = os.path.expanduser("~/.config")
+
+    if configRoot == "":
+        print("Default config locations don't exist. Ignoring external config.")
+        return ""
+
+    configHome = os.path.join(configRoot, "discord-rich-presence-plex")
+
+    if not os.path.isdir(configHome):
+        print("{} doesn't exist. Creating it.".format(configHome))
+        os.mkdir(configHome)
+
+    return configHome
+
+
 name = "Discord Rich Presence for Plex"
 version = "2.3.2"
 
 plexClientID = "discord-rich-presence-plex"
 discordClientID = "413407336082833418"
-configFilePath = "config.json"
-cacheFilePath = "cache.json"
-logFilePath = "console.log"
+configFilePath = os.path.join(get_config_dir(), "config.json")
+cacheFilePath = os.path.join(get_config_dir(), "cache.json")
+logFilePath = os.path.join(get_config_dir(), "console.log")
 
 isUnix = sys.platform in ["linux", "darwin"]
 processID = os.getpid()
