diff --git a/README.md b/README.md
index 04d8a61..7d50775 100644
--- a/README.md
+++ b/README.md
@@ -13,6 +13,8 @@ A Python script that displays your [Plex](https://www.plex.tv) status on [Discor
 
 Add your configuration(s) into the `plexConfigs` list on line 30.
 
+You can place your plexConfig options into a discordRichPresencePlex.conf inside of ~/.config.
+
 #### Example
 
 ```python
diff --git a/discordRichPresencePlex.conf b/discordRichPresencePlex.conf
new file mode 100644
index 0000000..4cdf72c
--- /dev/null
+++ b/discordRichPresencePlex.conf
@@ -0,0 +1 @@
+# plexConfig(serverName = "", username = "", password = "", token = "")
diff --git a/discordRichPresencePlex.py b/discordRichPresencePlex.py
old mode 100644
new mode 100755
index 2953014..ce88e5b
--- a/discordRichPresencePlex.py
+++ b/discordRichPresencePlex.py
@@ -1,3 +1,4 @@
+#!/usr/bin/env python3
 import asyncio
 import datetime
 import hashlib
@@ -11,6 +12,33 @@ import tempfile
 import threading
 import time
 
+def readConf():
+    confighome = ""
+    if 'APPDATA' in os.environ:
+        confighome = os.environ['APPDATA']
+    elif 'XDG_CONFIG_HOME' in os.environ:
+        confighome = os.environ['XDG_CONFIG_HOME']
+    elif os.path.isdir(os.path.expanduser('~/.config')):
+        confighome = os.path.expanduser('~/.config')
+
+    if confighome == "":
+        print("Config location couldn't be found. Ignoring external config.")
+        return []
+
+    conf = os.path.join(confighome, 'discordRichPresencePlex.conf')
+    if not os.path.isfile(conf):
+        print("Config file couldn't be found. Ignoring external config.")
+        return []
+
+    with open(conf) as f:
+        content = f.readlines()
+    content = [x.strip() for x in content]
+    confList = []
+    for x in content:
+        if not x.startswith("#"):
+            confList.append( eval(x) )
+    return confList
+
 class plexConfig:
 
 	extraLogging = True
@@ -29,6 +57,7 @@ class plexConfig:
 plexConfigs = [
 	# plexConfig(serverName = "", username = "", password = "", token = "")
 ]
+plexConfigs += readConf()
 
 class discordRichPresence:
 
