diff -up wine-5.11/dlls/ntdll/unix/signal_x86_64.c.omv~ wine-5.11/dlls/ntdll/unix/signal_x86_64.c
--- wine-5.11/dlls/ntdll/unix/signal_x86_64.c.omv~	2020-07-03 00:51:35.139727705 +0200
+++ wine-5.11/dlls/ntdll/unix/signal_x86_64.c	2020-07-03 01:32:41.452678766 +0200
@@ -1390,7 +1390,7 @@ static NTSTATUS libunwind_virtual_unwind
     unw_proc_info_t info;
     int rc;
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(____LIBUNWIND_CONFIG_H__) /* ____LIBUNWIND_CONFIG_H__ is a sign of LLVM libunwind */
     rc = unw_getcontext( &unw_context );
     if (rc == UNW_ESUCCESS)
         rc = unw_init_local( &cursor, &unw_context );
