diff --git a/configure.ac b/configure.ac
index 3c01867..25fe5a9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -257,10 +257,10 @@ dnl ######################
 dnl checking for iniparser
 dnl ######################
 
-AC_CHECK_LIB(iniparser,iniparser_load, have_iniparser=yes, have_iniparser=no)
+AC_CHECK_LIB(iniparser4,iniparser_load, have_iniparser=yes, have_iniparser=no)
     if [[ $have_iniparser = "yes" ]] ; then
-    LIBS="$LIBS -liniparser"
-      CPPFLAGS="$CPPFLAGS -I/usr/include/iniparser"
+    LIBS="$LIBS -liniparser4"
+      CPPFLAGS="$CPPFLAGS -I/usr/include/iniparser4"
     AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <iniparser.h>]],
       [[dictionary* ini;
       const char *keys[3];
